helix-core/src/main/java/org/apache/helix/participant/HelixStateMachineEngine.java

```diff
@@ -146,18 +147,29 @@ private void sendNopMessage() {
 
   @Override
   public void reset() {
-    logger.info("Resetting HelixStateMachineEngine");
+    loopStateModelFactories(stateModel -> {
+      stateModel.reset();
+      String initialState = _stateModelParser.getInitialState(stateModel.getClass());
+      stateModel.updateState(initialState);
+    }, "reset");
```

Comment:
NIT: Let's have the enum for "reset" and "syncState" instead of hard coding.

Reply:
It will never be too heavy for this. Hard code is not a good idea except in test cases. If we would like to change some logic in the future. We dont have the centralized place to trace back when these logic happens.

Reply:
How about this: we don't add this param just for logging purpose, instead, move the logging out to the specific method. It makes sense not to add the param and enum just for logging.